src/nodes.coffee

```diff
@@ -1202,25 +1312,30 @@ exports.Class = class Class extends Base
     args  = []
     o.classScope = func.makeScope o.scope
 
-    @hoistDirectivePrologue()
     @setContext name
-    @walkBody name, o
-    @ensureConstructor name
-    @addBoundFunctions o
-    @body.spaced = yes
-    @body.expressions.push lname
+    @hoistDirectivePrologue()
 
     if @parent
       superClass = new IdentifierLiteral o.classScope.freeVariable 'superClass', reserve: no
-      @body.expressions.unshift new Extends lname, superClass
+      @body.expressions.unshift new Assign \
```

Comment:
Rather than using the `extend` utility, we are compiling an `extends` in the `JsClass`. However, old `superReference` behaviour still depends on `__super__` (unless there's a way to look that up in vanilla JS).